projects/kubernetes-sigs/image-builder/build/setup_packer_configs.sh

```diff
@@ -160,12 +160,16 @@ if [ "$IMAGE_FORMAT" = "ova" ] && \
         if [ -z $PACKER_ACTIVE_INTERFACE ]; then
             echo "Finding interface for Packer temporary HTTP server"
             if [ "$(uname -s)" = "Linux" ]; then
-                if command -v route &> /dev/null; then
-                    PACKER_ACTIVE_INTERFACE=$(route |awk '/^default/{print $NF}')
+                # NOTE: not all Ubuntu variant comes "route" command by default, better to detect with "ip".
+                if command -v ip &> /dev/null; then
+                    PACKER_ACTIVE_INTERFACE=$(ip route show default | awk '/default/ {print $5}')
                 else
                     INTERFACES=($(ls /sys/class/net))
                     for interface in "${INTERFACES[@]}"; do
-                        if [ "$interface" = "eth0" ] || [ "$interface" = "en0" ] || [ "$interface" = "eno1" ]; then
+                        # Systemd v197 (or newer) comes with predictable interface name and under Ubuntu system on the Cloud, it would be "ens5".
+                        # - https://systemd.io/PREDICTABLE_INTERFACE_NAMES/
+                        # - https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/
+                        if [ "$interface" = "eth0" ] || [ "$interface" = "en0" ] || [ "$interface" = "eno1" ] || [ "$interface" = "ens5" ]; then
```

Comment:
I'm okay to keep adding to this list, but I think for readability's sake, it would be better to define an array and search from it.
```bash
predictable_interface_names=(
    eth0
    en0
    eno1
    ens5
)

if [[ ${predictable_interface_names[@]} =~ $interface ]]; then
...
```

Reply:
Sure thing, it's much easier to read ( or leaving comments for why we list such item )

Reply:
#### Script

```
$ cat test.sh
#!/bin/bash

interface=$1

predictable_interface_names=(
        "eth0"
        "en0"
        "eno1"
        "ens5"
)

if [[ ${predictable_interface_names[@]} =~ $interface ]]; then
    echo "found in array"
else
    echo "not in array"
fi
```

#### Test 1: ✅ 
```
$ bash -x test.sh eth0
+ interface=eth0
+ predictable_interface_names=("eth0" "en0" "eno1" "ens5")
+ [[ eth0 en0 eno1 ens5 =~ eth0 ]]
+ echo 'found in array'
found in array
```

#### Test 2: ✅ 
```
$ bash -x test.sh eth1
+ interface=eth1
+ predictable_interface_names=("eth0" "en0" "eno1" "ens5")
+ [[ eth0 en0 eno1 ens5 =~ eth1 ]]
+ echo 'not in array'
not in array
```

#### Test 3: ❌ 
```
% bash -x test.sh eth
+ interface=eth
+ predictable_interface_names=("eth0" "en0" "eno1" "ens5")
+ [[ eth0 en0 eno1 ens5 =~ eth ]]
+ echo 'found in array'
found in array
```

Reply:
Just a second, I can handle it, let me work on it...

Reply:
Updated.